amazon/ion/ion_py_objects.py

```diff
@@ -0,0 +1,586 @@
+
+    def to_event(self, event_type, field_name=None, in_struct=False, depth=None):
+        value = self
+        if isinstance(self, IonPyNull_new) or self.ion_type.is_container:
+            value = None
+
+        if in_struct:
+            if not isinstance(field_name, SymbolToken):
+                field_name = SymbolToken(field_name, 0 if field_name is None else None)
+        else:
+            field_name = None
+
+        return IonEvent(event_type, ion_type=self.ion_type, value=value, field_name=field_name,
+                        annotations=self.ion_annotations, depth=depth)
+
+
+class IonPySymbol_new(SymbolToken):
+    __name__ = 'IonPySymbol_new'
+    __qualname__ = 'IonPySymbol_new'
+    ion_type = IonType.SYMBOL
+
+    # a good signature: IonPySymbol(ion_type, symbol_token, annotation)
+    def __new__(cls, ion_type=IonType.SYMBOL, value=None, annotations=()):
+        v = super().__new__(cls, *value)
+        v.ion_annotations = annotations
+        return v
+
+    def __copy__(self):
+        args, kwargs = self._to_constructor_args(self)
+        value = self.__class__(*args, **kwargs)
+        value.ion_type = self.ion_type
+        value.ion_annotations = self.ion_annotations
+        return value
+
+    @staticmethod
+    def _to_constructor_args(st):
+        try:
+            args = (None, (st.text, st.sid, st.location), ())
+        except AttributeError:
+            args = (None, (st, None, None), ())
+        kwargs = {}
+        return args, kwargs
+
+    @classmethod
+    def from_event(cls, ion_event):
+        if ion_event.value is not None:
+            args, kwargs = cls._to_constructor_args(ion_event.value)
+        else:
+            args, kwargs = (None, None, ()), {}
+        value = cls(*args, **kwargs)
+        value.ion_type = ion_event.ion_type
+        value.ion_annotations = ion_event.annotations
+        return value
+
+    @classmethod
+    def from_value(cls, ion_type, value, annotations=()):
+        if value is None:
+            value = IonPyNull_new()
+        else:
+            args, kwargs = cls._to_constructor_args(value)
+            value = cls(*args, **kwargs)
+        value.ion_type = ion_type
+        value.ion_annotations = annotations
+        return value
+
+    def to_event(self, event_type, field_name=None, in_struct=False, depth=None):
+        value = self
+        if isinstance(self, IonPyNull_new) or self.ion_type.is_container:
+            value = None
+
+        if in_struct:
+            if not isinstance(field_name, SymbolToken):
+                field_name = SymbolToken(field_name, 0 if field_name is None else None)
+        else:
+            field_name = None
+
+        return IonEvent(event_type, ion_type=self.ion_type, value=value, field_name=field_name,
+                        annotations=self.ion_annotations, depth=depth)
+
+
+class IonPyList_new(list):
+    __name__ = 'IonPyList_new'
+    __qualname__ = 'IonPyList_new'
+    ion_type = IonType.LIST
```

Comment:
like with null I would not have the class attribute for List.

Reply:
Removed.